include/oneapi/tbb/blocked_range.h

```diff
@@ -108,6 +108,22 @@ class blocked_range {
         // only comparison 'less than' is required from values of blocked_range objects
         __TBB_ASSERT( !(my_begin < r.my_end) && !(r.my_end < my_begin), "blocked_range has been split incorrectly" );
     }
+  
+    // fill elements with their index values
+  void first_touch(std::vector<Value>& container) const {
```

Comment:
I think having a first_touch function as part of the range is not the right abstraction. This looks like an algorithm. Also, the first-touch code would be application-specific and likely achieved by combining a range, partitioner and algorithm (very likely parallel_for). So I would not expect an explicit first_touch algorithm either.